server/src/main/java/org/elasticsearch/search/aggregations/AggregatorBase.java

```diff
@@ -325,6 +325,11 @@ protected final BigArrays bigArrays() {
         return context.bigArrays();
     }
 
+
+    protected final BigArrays bigArraysForResults() {
```

Comment:
I'm not sure we actually need this.  Maybe we just have one BigArrays, and it outlives the context.  Otherwise, what's the use case for data that lives at collect time but doesn't extend to serialization time?